### PR TITLE
Better support for realizing scalar YAML values

### PR DESCRIPTION
**Pull request by maddenp-noaa**
**Synopsis**

Add YAML configuration to omit `...` end-of-stream marker when dumping scalar values. For example:

Old behavior:
```
(DEV-uwtools) ~/git/uwtools ∴ echo "{foo: {bar: baz}}" | uw config realize --input-format yaml --output-format yaml --key-path foo.bar
baz
...
```

New behavior:
```
(DEV-uwtools) ~/git/uwtools ∴ echo "{foo: {bar: baz}}" | uw config realize --input-format yaml --output-format yaml --key-path foo.bar
baz
```

This supports easier lookups of single values from input configs.

Works for e.g. namelist inputs, too:
```
(DEV-uwtools) ~/git/uwtools ∴ echo "&nl res = 768 /" | uw config realize --input-format nml --output-format yaml --key-path nl.res
768
```

**Type**

- [x] Enhancement (adds new functionality)

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/config/formats/yaml.py [hunk 1]
```diff
@@ -145,3 +141,52 @@ def get_format() -> str:
         Returns the config's format name.
         """
         return FORMAT.yaml
+
+    # Private methods
```
Comment (maddenp-noaa) [inline src/uwtools/config/formats/yaml.py:145]: I relocated some YAML-related code from `uwtools.config.support` to the `YAMLConfig` class. I half expected a circular-dependency issue to arise in the refactor, and assumed that that's the reason the code lived where it did, but was pleasantly surprised to find no issues incorporating it into the class where it seems like it belongs.

File: src/uwtools/config/formats/yaml.py [hunk 2]
```diff
@@ -145,3 +141,52 @@ def get_format() -> str:
...
+    Write YAML without ...
+
+    end-of-stream marker.
+    """
+    self.write_plain_base(*args, **kwargs)
+    self.open_ended = False
+
+
+setattr(yaml.emitter.Emitter, "write_plain_base", yaml.emitter.Emitter.write_plain)
+setattr(yaml.emitter.Emitter, "write_plain", _write_plain_open_ended)
```
Comment (maddenp-noaa) [inline src/uwtools/config/formats/yaml.py:192]: These two `setattr()` calls and the `_write_plain_open_ended()` function they refer to are responsible for removing the `...` end-of-stream marker from the YAML-scalar output.

File: src/uwtools/tests/config/test_tools.py [hunk 3]
```diff
@@ -347,6 +347,16 @@ def test_realize_config_remove_yaml_to_yaml_subtree(tmp_path):
...
 
 
+def test_realize_config_scalar_value(capsys):
+    stdinproxy.cache_clear()
+    tools.realize_config(
+        input_config=YAMLConfig(config={"foo": {"bar": "baz"}}),
+        output_format="yaml",
+        key_path=["foo", "bar"],
+    )
+    assert capsys.readouterr().out.strip() == "baz"
```
Comment (maddenp-noaa) [inline src/uwtools/tests/config/test_tools.py:357]: A test for the new behavior. Previously, the RHS of the `assert` statement would have been `baz\n...`.

**Review by christinaholtNOAA — APPROVED** [review]
LGTM!

File: src/uwtools/config/formats/yaml.py (hunk 2 above)
Comment (WeirAE) [reply src/uwtools/config/formats/yaml.py:192]: aha, nice!

**Review by WeirAE — APPROVED** [review]
LGTM

**Pull request merged**